packages/table/src/table.tsx

```diff
@@ -1827,10 +1827,11 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         // change selection to match new focus cell location
         const newSelectionRegions = [Regions.cell(newFocusedCell.row, newFocusedCell.col)];
         const { selectedRegionTransform } = this.props;
-        if (selectedRegionTransform != null) {
-            newSelectionRegions.forEach(region => selectedRegionTransform(region, undefined));
-        }
-        this.handleSelection(newSelectionRegions);
+        const transformedSelectionRegions =
+            selectedRegionTransform != null
+                ? newSelectionRegions.map(region => selectedRegionTransform(region, undefined))
```

Comment:
this `undefined` breaks the type contract. blocker for this feature.

Reply:
Might be overkill but would it be acceptable to create a fake `MouseEvent` that simulates a click on the middle of the cell that the focus is being moved to?

Reply:
this code is inside an event handler so we do have an event, it's just not a MouseEvent. best solution may be to change type to `MouseEvent | FocusEvent`

Reply:
We have a `KeyboardEvent` there, not a `FocusEvent`, but I see your point. That would be a major version bump, though - so this has to wait until Blueprint 4, right?

Reply:
oh right my bad it is a keyboard event. 

honestly this table library will never have a v4. we're working on a complete rewrite internally that will replace this implementation completely. it's going to be so much better but also quite different.

Reply:
i'm willing to merge this change and declare it a bug fix, not an API break. makes sense for keyboard events to go through this flow just like mouse events. typescript users will see compile errors if they're relying on `MouseEvent` properties so at least it won't be silent.

Reply:
Changed the typing of `ISelectedRegionTransform` in 6a35c0a0 - this version now also forwards the `KeyboardEvent` to the region transform function.